src/core/shape_inference/include/ctc_greedy_decoder_seq_len_shape_inference.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/ctc_greedy_decoder_seq_len.hpp>
+
+namespace ov {
+namespace op {
+namespace v6 {
+
+template <class T>
+void shape_infer(const CTCGreedyDecoderSeqLen* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 2 || input_shapes.size() == 3 && output_shapes.size() == 2);
```

Comment:
Looks like we don't check `output_shapes` size in case when `input_shapes` size is 2
```suggestion
    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 2 || input_shapes.size() == 3) && output_shapes.size() == 2);
```